A_Star/A_Star.cpp

```diff
@@ -0,0 +1,224 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <queue>
+#include <unordered_map>
+
+using namespace std;
+
+struct Node
+{
+    string name;
+    int currentCost;
+    bool hasVisited;
+    unordered_map<Node*, int> neighbors;
```

Comment:
will be more memory efficient with std::vector, since you aren't looking up the neighbor by the key but just simply iterating thru all of them.  unorder_map has a cost of a hashtable (for the look up)